recipes/libidn/all/conanfile.py

```diff
@@ -97,10 +99,12 @@ def _configure_autotools(self):
             "--disable-nls",
             "--disable-rpath",
         ]
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
+        autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
+        return autotools
 
     def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```

Comment:
It would be better to have it in a separate method,
```

    def _patch_sources(self):
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```

Reply:
it's fine here when patches is limited to this logic